services/management/dbaas/kubernetes_server.go

```diff
@@ -52,19 +53,46 @@ func (k kubernetesServer) ListKubernetesClusters(ctx context.Context, _ *dbaasv1
 		return nil, err
 	}
 
+	var wg sync.WaitGroup
 	clusters := make([]*dbaasv1beta1.ListKubernetesClustersResponse_Cluster, len(kubernetesClusters))
 	for i, cluster := range kubernetesClusters {
-		clusters[i] = &dbaasv1beta1.ListKubernetesClustersResponse_Cluster{
-			KubernetesClusterName: cluster.KubernetesClusterName,
-		}
+		i := i
+		cluster := cluster
+		wg.Add(1)
+		go func() {
+			defer wg.Done()
+			clusters[i] = &dbaasv1beta1.ListKubernetesClustersResponse_Cluster{
+				KubernetesClusterName: cluster.KubernetesClusterName,
+				Operators: &dbaasv1beta1.Operators{
+					Xtradb: new(dbaasv1beta1.Operator),
+					Psmdb:  new(dbaasv1beta1.Operator),
+				},
+			}
+			resp, e := k.dbaasClient.CheckKubernetesClusterConnection(ctx, cluster.KubeConfig)
+			if e != nil {
+				clusters[i].Status = dbaasv1beta1.KubernetesClusterStatus_KUBERNETES_CLUSTER_STATUS_UNAVAILABLE
+			}
+
+			if resp == nil {
```

Comment:
Looks like this case only possible with error https://github.com/percona-platform/dbaas-controller/pull/106/files#diff-611c99343ad28b77556d6ca5ebfe5d6b63a2f6e4cbaf38927a605018c220351dR50
 
so may be we can do:

```
if e != nil {
    clusters[i].Status = dbaasv1beta1.KubernetesClusterStatus_KUBERNETES_CLUSTER_STATUS_UNAVAILABLE
    return 
}
```